examples/ansible/roles/alp_reboot/tasks/start.yml

```diff
@@ -0,0 +1,41 @@
+- name: Enable lingering for the user
+  ansible.builtin.command:
+    cmd: "loginctl enable-linger {{ ansible_user }}"
+
+- name: Check if the service unit file exists
```

Comment:
What happens if two different playbooks are running on the system, and want to configure resumption after reboot?

With this logic flow, the second playbook to run would remove the first playbook's service definition...

I understand that this shouldn't really be happening, but what if someone configured a playbook to run after a reboot, that in turn needs to trigger a reboot, and then someone else, or even the same user, runs a playbook after the first reboot, but before the second reboot, or before the resumed playbook got to the cleanup phase? In such a case we could "break" things due to the racing actions...

Reply:
Note that one way of dealing with this would be to define an appropriate service template, i.e. a service unit with an `@` on the end of the name, and require the use of the role to provide a service instance identifier, and then configure specific instances of that service for the provide instance name.

That would partially mitigate things, and would even allow you to define multiple playbooks to be run after a reboot...

However, there would still be the potential for those playbooks to race against each other as they could be triggered at the same time.

Reply:
Another alternative would be to define a persistent generic service definition that checks a particular directory location, e.g. /etc/ansible-resume-playbooks/, for resumption config files, which will be processed in order by the service, which avoid the risk of racing resume playbooks.

However, you would need to ensure that if a resumed playbook itself added an entry, to be resumed after the next reboot, then it should not be resumed until after the next reboot.